slack_sdk/web/async_client.py

```diff
@@ -2992,6 +2997,166 @@ async def files_upload(
+        content: Optional[Union[str, bytes]] = None,
+        title: Optional[str] = None,
+        alt_txt: Optional[str] = None,
+        snippet_type: Optional[str] = None,
+        # To upload multiple files at a time
+        upload_files: Optional[List[Dict[str, Any]]] = None,
+        channel: Optional[str] = None,
+        initial_comment: Optional[str] = None,
+        thread_ts: Optional[str] = None,
+        **kwargs,
+    ) -> AsyncSlackResponse:
+        """This wrapper method provides an easy way to upload files using the following endpoints:
+        * step1: https://api.slack.com/methods/files.getUploadURLExternal
+        * step2: "https://files.slack.com/upload/v1/..." URLs returned from files.getUploadURLExternal API
+        * step3: https://api.slack.com/methods/files.completeUploadExternal
+        """
+        if file is None and content is None and upload_files is None:
+            raise e.SlackRequestError("Any of file, content, and upload_files must be specified.")
+        if file is not None and content is not None:
+            raise e.SlackRequestError("You cannot specify both the file and the content argument.")
+
+        # deprecated arguments:
+        channels, filetype = kwargs.get("channels"), kwargs.get("filetype")
+
+        if channels is not None:
+            warnings.warn(
+                "Although the channels parameter is still supported for smooth migration from legacy files.upload, "
+                "we recommend using the new channel parameter with a single str value instead for more clarity."
+            )
+            if (isinstance(channels, (list, Tuple)) and len(channels) > 1) or (
+                isinstance(channels, str) and len(channels.split(",")) > 1
+            ):
+                raise e.SlackRequestError(
+                    "Sharing files with multiple channels is no longer supported in v2. "
+                    "Share files in each channel separately instead."
+                )
+        if filetype is not None:
+            warnings.warn("The filetype parameter is no longer supported. Please remove it from the arguments.")
+
+        # step1: files.getUploadURLExternal per file
+        files: List[Dict[str, Any]] = []
+        if upload_files is not None:
+            for f in upload_files:
+                files.append(_to_v2_upload_file_item(f))
+        else:
+            f = _to_v2_upload_file_item(
+                {
+                    "filename": filename,
+                    "file": file,
+                    "content": content,
+                    "title": title,
+                    "alt_txt": alt_txt,
+                    "snippet_type": snippet_type,
+                }
+            )
+            files.append(f)
+
+        for f in files:
+            url_response = await self.files_getUploadURLExternal(
+                filename=f.get("filename"),
+                length=f.get("length"),
+                alt_txt=f.get("alt_txt"),
```

Comment:
Related to above comment! As `f` in files are the output of `_to_v2_upload_file_item` this would return None.